### PR TITLE
Added new possible urls and supporting functions for bookstore API.

### PR DESCRIPTION
**Pull request by kmb4577**

### CODE REVIEW

File: API/API.php [hunk 1]
```diff
@@ -300,44 +300,152 @@ function postCourse()
...
 	}
 }
 
 //Define Functions Here
+function createBook($isbn, $title, $publisher_id, $price, $thumbnail_url, $available, $count)
+{
+	echo $isbn;
+	try 
+		{
+			$sqlite = new SQLite3(__DIR__.DIRECTORY_SEPARATOR."SWEN344DB.db");
```
Comment (ZacharyMoran) [inline API/API.php]: Are you just using this db for testing? We have the shared db with the scripts to run all the .sql files all set up.

File: API/API.php [hunk 2]
```diff
@@ -300,44 +300,152 @@ function postCourse()
...
+					logError("getBook ~ Required isbn parameter was not submitted correctly.");
+					return ("getBook book isbn parameter was not submitted correctly.");
+				}
 				// return "Missing " . $_GET["param-name"]
 			case "getSectionBook":
-				// if has params
-				return getSectionBook();
-				// else
-				// return "Missing " . $_GET["param-name"]
-			case "postBook":
```
Comment (ZacharyMoran) [inline API/API.php]: are you not providing a way to add books to the system? How will our group let professors add new books if its not already in the db?

File: API/API.php [hunk 3]
```diff
@@ -300,44 +300,152 @@ function postCourse()
...
 }
 
-function getSectionBook()
+function getBook($isbn)
 {
 	return "TODO";
 }
 
-function postBook()
+function getSectionBook($isbn, $section_id)
```
Comment (ZacharyMoran) [inline API/API.php]: ideally this function should only need the section id as a param and would only ever be used as "getSectionBooks" plural

File: API/API.php (hunk 2 above)
Comment (ZacharyMoran) [inline API/API.php]: nvm i did neglected to read the create books function i apologize

File: API/API.php (hunk 1 above)
Comment (kmb4577) [inline API/API.php]: yes

File: API/API.php (hunk 3 above)
Comment (kmb4577) [inline API/API.php]: ok. I'll change that in the next pull request.

File: API/API.php (hunk 1 above)
Comment (roda0313) [inline API/API.php]: If this is jut for testing then it should be commented out before commit!

Comment (kmb4577) [inline API/API.php]: got that, thanks

File: API/API.php [hunk 4]
```diff
@@ -300,44 +300,152 @@ function postCourse()
...
 	// Define the possible Book Store function URLs which the page can be accessed from
-	$possible_function_url = array("getBook", "getSectionBook", "postBook");
+	$possible_function_url = array("createBook", "updateBook", "getBook", "getSectionBook");
 
 	if (isset($_GET["function"]) && in_array($_GET["function"], $possible_function_url))
 	{
 		switch ($_GET["function"])
 		{
+			case "createBook":
+				if (isset($_GET["isbn"]) &&
```
Comment (roda0313) [inline API/API.php]: You guys might want to take another look at the different between GET and POST requests. If you are creating a new book using a GET request, then anyone could just bypass your form and any validation and use the URL to directly put parameters into the API call

**Pull request closed without merging**